packages/regenerator-transform/src/visit.js

```diff
@@ -150,6 +150,18 @@ exports.getVisitor = ({ types: t }) => ({
         path.addComment("leading", "#__PURE__");
       }
 
+      const insertedLocs = emitter.getInsertedLocs();
+
+      path.traverse({
```

Comment:
this is a hack-ish solution for "intentional" duplicates, cant rly clone them when inserting to AST because they need to be the same object for [`mark`](https://github.com/facebook/regenerator/blob/92f5f8534d5005cf56d894a782e12b842628b5ed/packages/regenerator-transform/src/emit.js#L68) to take effect on all inserted instances

This whole thing with updating those nodes after they get inserted simplifies the transform. Rewriting it to avoid duplicates has too big cost and would be really hard.

I've considered tracking all insertions but it would complicate things quite a bit because there is no single place for AST insertions (kinda general rule for any babel transform 😉 )

So I've decided to just "fix" the transformed subtree after whole transform takes place.